src/cpp/session/modules/rmarkdown/SessionRMarkdown.cpp

```diff
@@ -1433,32 +1433,160 @@ Error executeRcppEngineChunk(const std::string& docId,
-   return Success();
+   // forward declare error
+   Error error = Success();
+   
+   // always ensure we emit a 'execution complete' event on exit
+   ChunkExecCompletedScope execScope(docId, chunkId);
+   
+   // prepare chunk directory
+   FilePath outputPath = notebook::chunkOutputPath(
+            docId,
+            chunkId,
+            notebook::ContextExact);
+
+   error = outputPath.removeIfExists();
+   if (error)
+      LOG_ERROR(error);
+
+   error = outputPath.ensureDirectory();
+   if (error)
+      LOG_ERROR(error);
+   
+   // prepare to write chunk cache output
+   FilePath target = notebook::chunkOutputFile(docId, chunkId, kChunkOutputText);
+   
+   // capture console output, error
+   boost::signals::scoped_connection consoleHandler =
+         module_context::events().onConsoleOutput.connect(
+            boost::bind(sourceCppConsoleOutputHandler,
+                        _1,
+                        _2,
+                        target));
+   
+   // write input code to cache
+   error = notebook::appendConsoleOutput(
+            kChunkConsoleInput,
+            code,
+            target);
+   if (error)
+      LOG_ERROR(error);
+   
+   // write code to file
+   FilePath tempFile = module_context::tempFile("stan-", ".stan");
+   error = writeStringToFile(tempFile, code);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return error;
+   }
+   RemoveOnExitScope removeOnExitScope(tempFile, ERROR_LOCATION);
+   
+   // evaluate engine options (so we can pass them through to stan call)
+   r::sexp::Protect protect;
+   SEXP engineOptsSEXP = R_NilValue;
+   if (chunkOptions.count("engine.opts"))
+   {
+      error = r::exec::evaluateString(
+               chunkOptions.at("engine.opts"),
+               &engineOptsSEXP,
+               &protect);
+      if (error)
+         LOG_ERROR(error);
+   }
+   
+   // construct call to 'stan_model'
+   r::exec::RFunction fStanEngine("rstan:::stan_model");
+   std::vector<std::string> engineOptsNames;
+   error = r::sexp::getNames(engineOptsSEXP, &engineOptsNames);
+   if (error)
+   {
+      LOG_ERROR(error);
+      return error;
+   }
+   
+   // build parameters
+   std::string modelName;
+   for (std::size_t i = 0, n = r::sexp::length(engineOptsSEXP); i < n; ++i)
+   {
+      // skip 'x' engine option (this is the variable we wish to assign to
+      // after evaluating the stan model)
+      if (engineOptsNames[i] == "x")
+      {
+         modelName = r::sexp::asString(VECTOR_ELT(engineOptsSEXP, i));
+         continue;
+      }
+      
+      fStanEngine.addParam(
+               engineOptsNames[i],
+               VECTOR_ELT(engineOptsSEXP, i));
+   }
+   
+   // if the 'file' option was not set, set it explicitly
+   if (!core::algorithm::contains(engineOptsNames, "file"))
+      fStanEngine.addParam("file", tempFile.absolutePath());
+   
+   // evaluate stan_model call
+   SEXP stanModelSEXP = R_NilValue;
+   error = fStanEngine.call(&stanModelSEXP, &protect);
```

Comment:
If an error occurs here (e.g. because the stan code is invalid or the rstan package isn't installed) then the user needs to make sure they see it. We should probably do the exact same thing we do for C++ here and call the sourceCppConsoleOutputHandler directly with the error message (which you can get via `r::endUserErrorMessage(error)`)